scripts/build_binary_and_layer_dockerized.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+set -e
+
+if [ -z "$VERSION" ]; then
+    echo "Extension version not specified"
+    echo ""
+    echo "EXITING SCRIPT."
+    exit 1
+fi
+
+AGENT_PATH="../datadog-agent"
+EXTENSION_DIR=".layers"
+TARGET_DIR=$(pwd)/../$EXTENSION_DIR
+
+# Move into the root directory, so this script can be called from any directory
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd $SCRIPTS_DIR/..
+
+# First prepare a folder with only *mod and *sum files to enable Docker caching capabilities
+mkdir -p ./scripts/.src ./scripts/.cache
```

Comment:
This might touch the go.mod and go.sum files for every build. I'm not sure, but that might force the Docker file to rebuild all it's layers from the point you important the file. I'm not sure if docker uses last touched time on files, or whether it does a diff.